### ISSUE TITLE
Template data doesn't get rendered in topology file

### ISSUE THREAD

**Issue by lukas-bednar** [CentOS-PaaS-SIG/linchpin#576]
linchpin version 1.5.3

I have following topology file where I use list of variables: image, executor_name, network, etc
```yaml
# linchpin-ws/topologies/my-cluster.yml 
---
topology_name: my_cluster
resource_groups:
  - resource_group_name: openstack
    resource_group_type: openstack
    resource_definitions:
      - name: "{{ executor_name }}-master"
        type: os_server
        flavor: w1.large
        image: "{{ image }}"
        count: 1
        keypair: "{{ keypair }}"
        networks:
          - "{{ network }}"
        fip_pool: "{{ fip_pool }}"
        security_groups: "{{ security_group }}"
      - name: "{{ executor_name }}-node"
        type: os_server
        flavor: w1.large
        image: "{{ image }}"
        count: 2
        keypair: "{{ keypair }}"
        networks:
          - "{{ network }}"
        fip_pool: "{{ fip_pool }}"
        security_groups: "{{ security_group }}"
    credentials:
      filename: clouds.yml
      profile: "{{ os_cloud }}"
```

I want to pass these variables from template-data file as following
```yaml
# linchpin-ws/Data.my_cluster.yml 
---
executor_name: my-executor-lbednar
keypair: my-executor-lbednar-net-key
network: my-executor-lbednar-net
security_group: my-executor-lbednar-secgroup
image: rhel-7.4-server-x86_64-latest
fip_pool: 10.8.240.0
os_cloud: centralci
```
Linchpin fails to find `image` parameter, because `image: "{{ image }}` in topology file was rendered into `image: ""` so it looks like that template-data are not used to render topology file.
```
$ linchpin -vvv -w linchpin-ws/ --creds-path /home/my-qe-jenkins/.config/openstack -d @Data.my_cluster.yml up my_cluster
TASK [openstack : provision/teardown os_server resources with provided auth] ***
task path: /home/my-qe-jenkins/jenkins/workspace/deploy-my-cluster-lbednar/.venv/lib/python2.7/site-packages/linchpin/provision/roles/openstack/tasks/provision_os_server.yml:2
fatal: [localhost]: FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_args": {
            "api_timeout": 99999, 
            "auth": {
                "changed": true, 
                "failed": false, 
                "gid": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "group": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "mode": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "output": {
                    "clouds": {
                        "centralci": {
                            "auth": {
                                "auth_url": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                                "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                                "project_name": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                                "username": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
                            }
                        }
                    }
                }, 
                "owner": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "params": {
                    "cred_path": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                    "cred_type": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                    "driver": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                    "filename": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
                }, 
                "path": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "secontext": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "size": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "state": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER", 
                "uid": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER"
            }, 
            "auth_type": null, 
            "auto_ip": true, 
            "availability_zone": null, 
            "boot_from_volume": false, 
            "boot_volume": null, 
            "cacert": null, 
            "cert": null, 
            "cloud": null, 
            "config_drive": false, 
            "count": 1, 
            "delete_fip": false, 
            "endpoint_type": "public", 
            "flavor": "w1.large", 
            "flavor_include": null, 
            "flavor_ram": null, 
            "floating_ip_pools": [
                ""
            ], 
            "floating_ips": null, 
            "image": "", 
            "image_exclude": "(deprecated)", 
            "key": null, 
            "key_name": "", 
            "meta": null, 
            "name": "-master", 
            "network": null, 
            "nics": [
                "net-name="
            ], 
            "region_name": null, 
            "reuse_ips": true, 
            "scheduler_hints": null, 
            "security_groups": [
                ""
            ], 
            "state": "present", 
            "terminate_volume": false, 
            "timeout": 1200, 
            "userdata": null, 
            "verify": null, 
            "volume_size": false, 
            "volumes": [], 
            "wait": true
        }
    }, 
    "msg": "Parameter 'image' or 'boot_volume' is required if state == 'present'"
}
```
Bottom line, it seems to be regression in new version, because this functionality used to work fine in previous version.

**Comment by jaypoulz** [comment]
I encountered this as well. Herlo has already fixed it in the develop branch.

Immediate workaround is to inline the topology in your PinFile.

**Comment by herlo** [comment]
Let me verify this works in the latest code.

**Comment by herlo** [comment]
I can indeed verify this has been fixed in the lastest code.

**Comment by herlo** [comment]
I'll be adding a test case as part of #528

**Comment by lukas-bednar** [comment]
I verified that this was fixed in 1.5.4, thank you.
